system/Database/SQLSRV/Builder.php

```diff
@@ -618,4 +618,137 @@ public function get(?int $limit = null, int $offset = 0, bool $reset = true)
 
         return $result;
     }
+
+    /**
+     * Generates a platform-specific upsertBatch string from the supplied data
+     *
+     * @throws DatabaseException
+     */
+    protected function _upsertBatch(string $table, array $keys, array $values): string
+    {
+        $sql = $this->QBOptions['sql'] ?? '';
+
+        // if this is the first iteration of batch then we need to build skeleton sql
+        if ($sql === '') {
+            $fullTableName = $this->getFullName($table);
+
+            $constraints = $this->QBOptions['constraints'] ?? [];
+
+            $tableIdentity = $this->QBOptions['tableIdentity'] ?? '';
+            $sql           = "SELECT name from syscolumns where id = Object_ID('" . $table . "') and colstat = 1";
+            if (($query = $this->db->query($sql)) === false) {
+                throw new DatabaseException('Failed to get table identity'); // @codeCoverageIgnore
```

Comment:
Why is `@codeCoverageIgnore` added?

Reply:
because the query shouldn't really ever fail.. only if for some reason the database is not available.

Reply:
This code is not unreachable because databases could be unavailable and that is what the code is for. 
It is simply not tested by PHPUnit and the tag `@codeCoverageIgnore` should be removed.

Then the code with the tag added after throwing exception is simply not tested either.

I don't think it is good practice to add the tag to make it look like coverage has increased 
just because it is difficult to write test code.

Reply:
I removed it